### ISSUE TITLE
Using model forms

### ISSUE THREAD

**Issue by Macodom** [francescortiz/image#8]
Sorry, I don't think it is really a issue, is more a question about functionality.

This module only works for Django Admin? Because I wanted to use it in models forms and calling the attributes like I would do with forms and templates but it didn't work, I'm using also S3 so I would like a better explanation about config settings variables if it is possible.

**Comment by francescortiz** [comment]
Your questions are completely fair.

The image processing works everywhere, but I never used the form widget outside of the admin form so I never took care of making it work outside of the administration pages. Making it work is pretty easy, though, but I want to migrate to new Django 1.11 widget HTML templates. I will update the module, In order to have the widget work anywhere.

On the S3 side, S3 is very slow to check for existance of files, so I use a filesystem storage that caches locally the existance of files in order to be able to know quickly if a file has already been generated. Also, you need to add `autogen=true` to all images in order to have the images pregenerated, since S3 lives in an external server and we cannot generated the images on demand.

I will take care of this things between today and tomorrow.

**Comment by francescortiz** [comment]
I added LocallyMirroredS3BotoStorage to the git repository. You can install it with

    pip install git+https://github.com/francescortiz/image.git

I laso documented the `S3_MIRROR_ROOT` that is used by `image.storages.LocallyMirroredS3BotoStorage`. I haven't tested it for some time now, so it might need to get updated. Feel free to ask for feedback or make a pull request.

**Comment by Macodom** [comment]
Thanks for the update, so this only works like an upgrade of s3 performance right? I'll check it today, the thing about making work outside admin is still going?

**Comment by francescortiz** [comment]
Yes, the S3 storage just is a huge performance upgrade. The ImageCenterWidget working only in the admin is in the workshop, but can be tested if you use an account that has the is_staff flag set to true. Regarding multiple images, it works the same whether it is a regular form or an inline form.
